lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -74,6 +76,11 @@ def index(self, trans, **kwd):
             encoded_id = trans.security.encode_id(wf_sa.stored_workflow.id)
             item['url'] = url_for( 'workflow', id=encoded_id )
             item['owner'] = wf_sa.stored_workflow.user.username
+            item['latest_workflow_steps'] = len( wf_sa.stored_workflow.latest_workflow.steps )
+            item['user_email'] = wf_sa.stored_workflow.user.email
+            item['user_name'] = wf_sa.stored_workflow.user.username
+            item['slug'] = wf_sa.stored_workflow.slug
+            item['published'] = wf_sa.stored_workflow.published            
```

Comment:
As above, this is not necessary, `'published'` field is already filled by the `to_dict` method.